components/raftstore/src/coprocessor/region_info_accessor.rs

```diff
@@ -563,46 +565,61 @@ impl RegionCollector {
     }
 
     /// Used by the in-memory engine (if enabled.)
-    /// If `count` is 0, return all the regions for which this node is the
-    /// leader. Otherwise, return the top `count` regions from
-    /// `self.region_activity`. Top regions are determined by comparing
-    /// `read_keys + written_keys` in each region's most recent region stat.
-    /// If count > 0 and size of `self.region_activity` is `> 0`, remove but the
-    /// top `count` regions from `self.region_activity`.
+    /// If `count` is 0, return all the regions.
+    ///
+    /// Otherwise, return the top `count` regions for which this node is the
+    /// leader from `self.region_activity`. Top regions are determined by
+    /// comparing `read_keys` in each region's most recent
+    /// region stat.
     ///
     /// Note: this function is `O(N log(N))` with respect to size of
     /// region_activity. This is acceptable, as region_activity is populated
     /// by heartbeats for this node's region, so N cannot be greater than
     /// approximately `300_000``.
-    pub fn handle_get_top_regions(&mut self, count: usize, callback: Callback<Vec<Region>>) {
+    pub fn handle_get_top_regions(&mut self, count: usize, callback: Callback<TopRegions>) {
```

Comment:
Should we clean the `region_activity` after every time called the `handle_get_top_regions` since there is no where stale old regions or follower regions for `region_activity` which may cause the `region_activity` grow infinitely.

Reply:
Good call, re: clean this.